### DISCUSSION TITLE
Conditional assert

### DISCUSSION THREAD

**Discussion by antback** [ovh/venom#682] [General]
Hi guys,

I'm trying to test an API that returns and array with different objects like this:

```json
{
    "data": [
        {
            "result": {
                "field1": "name",
                "field2": "name"
            },
        },
        {
            "result": {
                "field3": "name",
                "field4": "name"
            },
        }
    ]
}
```

You can see that the attributes of the first object are different from the second one. 

Is there any way of asserting only if something is present? Making conditional asserts?

Thanks a lot!

**Comment by yesnault** [comment]
Hi @antback, no, it's not possible with the built-in assertion engine. I think that the easiest way to do that is to put the json into a variable or file, then execute a step with the executor 'exec' to check the value with jq or something else.
